### PR TITLE
DOCS: Add RetroWave OPL3 documentation

### PR DESCRIPTION
**Pull request by NMIError**
This adds info to the documentation on the RetroWave OPL3 and how to configure it.

### CODE REVIEW

**Review by cadihowley — COMMENTED** [review]
Looks good! I’d suggest turning (the keys start with ``opl2lpt_`` and ``retrowaveopl3_``) into a standalone sentence i.e. removing the parentheses

File: doc/docportal/advanced_topics/configuration_file.rst [hunk 1]
```diff
@@ -363,6 +364,17 @@ There are many recognized configuration keys. In the table below, each key is ei
...
+	Specifies how the RetroWave OPL3 is connected:
+	- serial (connected to a USB port using a PotatoPi)
+	- spi (connected as a HAT using SPI) "
+		":ref:`retrowaveopl3_disable_buffer <adlib>`",boolean,false,
+		":ref:`retrowaveopl3_port <adlib>`",string,,"
+	Specifies the serial port that the RetroWave OPL3 is connected to.
+	For example:
+	- COM3
+	- ttyACM2 "
+		":ref:`retrowaveopl3_spi_cs <adlib>`",string,,Specifies the GPIO chip and line that the RetroWave OPL3 is connected to. Use the format <chip>,<line>.
```
Comment (cadihowley) [inline doc/docportal/advanced_topics/configuration_file.rst:377]: Add quotation marks to the description “Specifies the GPIO chip…” to fix the formatting in the table - at the moment it’s reading the comma and moving ```<line>``` to a new cell

File: doc/docportal/advanced_topics/configuration_file.rst [hunk 2]
```diff
@@ -363,6 +364,17 @@ There are many recognized configuration keys. In the table below, each key is ei
 	- 2gs
 	- atari
 	- macintosh "
+		":ref:`retrowaveopl3_bus <adlib>`",string,,"
+	Specifies how the RetroWave OPL3 is connected:
+	- serial (connected to a USB port using a PotatoPi)
```
Comment (cadihowley) [inline doc/docportal/advanced_topics/configuration_file.rst:369]: Seems to be an issue with the formatting of the bullet points but to be honest I can’t see why at first glance. I’ll have another look and try to figure it out tomorrow :)

**Comment by NMIError** [comment]
I've resolved the first two issues. Let me know what needs to be done for the bus bullet points. I'm not set up to compile the docs site, so I can't really see the results of the changes I make. :-)

**Comment by NMIError** [comment]
Thanks! If you are happy with the changes now, I'll merge the PR.

**Comment by cadihowley** [comment]
Done! Thanks again for updating the docs!!

**Comment by NMIError** [comment]
Thank you for your help!

**Pull request merged**